src/vm/arm/asmhelpers.S

```diff
@@ -441,7 +444,7 @@ LOCAL_LABEL(UMThunkStub_WrongAppDomain):
         ldr                 r2, [r3, #UMThunkMarshInfo__m_cbActualArgSize]
         cbz                 r2, LOCAL_LABEL(UM2MThunk_WrapperHelper_ArgumentsSetup)
 
-        add                 r0, r5, #UMThunkStub_StackArgs // Source pointer
+        add                 r0, r5, #UMThunkStub_StackArgsOffset // Source pointer
```

Comment:
I just suspected that ``UM2MThunk_WrapperHelper`` may take the value ``r7`` as its first argument, but I find no such code until now. 

I'll change that line as you suggested.